lib/doorkeeper/grant_flow.rb

```diff
@@ -0,0 +1,54 @@
+require 'doorkeeper/grant_flow/flow'
+
+module Doorkeeper
+  module GrantFlow
+    mattr_accessor :registered_flows
+    self.registered_flows = {}
+
+    module_function
+
+    def register(name_or_flow, options = {})
+      if name_or_flow.is_a? Doorkeeper::GrantFlow::Flow
+        registered_flows[name_or_flow.name.to_sym] = name_or_flow
```

Comment:
Hmm… I'm trying to remember what led me to add it in the first place. I think the idea was to let plugins do more complex matching by subclassing `Doorkeeper::GrantFlow::Flow` and registering a pre-built flow from their subclass. I'm happy to drop it if we don't anticipate needing that flexibility.